test/puppetlabs/puppetdb/scf/migrate_test.clj

```diff
@@ -1437,7 +1438,209 @@
                        :logs (sutils/munge-json-for-storage [{:bar "baz"}])})
         (apply-migration-for-testing! 75)
         (is (= "agent" (-> (query-to-vec "select * from reports")
-                           first :report_type)))))))
+                           first :report_type)))
+        (is (= {:index-diff nil,
+                :table-diff
+                [{:left-only nil,
+                  :right-only
+                  {:numeric_scale nil,
+                   :column_default "'agent'::text",
+                   :character_octet_length 1073741824,
+                   :datetime_precision nil,
+                   :nullable? "NO",
+                   :character_maximum_length nil,
+                   :numeric_precision nil,
+                   :numeric_precision_radix nil,
+                   :data_type "text",
+                   :column_name "report_type",
+                   :table_name "reports"},
+                  :same nil}
+                 {:left-only nil,
+                  :right-only
+                  {:numeric_scale nil,
+                   :column_default "'agent'::text",
+                   :character_octet_length 1073741824,
+                   :datetime_precision nil,
+                   :nullable? "NO",
+                   :character_maximum_length nil,
+                   :numeric_precision nil,
+                   :numeric_precision_radix nil,
+                   :data_type "text",
+                   :column_name "report_type",
+                   :table_name "reports_20211115z"},
```

Comment:
These table names are going to be dependent on what day the test are run on due to the logic over in [`migrate.clj`](https://github.com/puppetlabs/puppetdb/blob/10435954d814f737a94752d264595d4a7e88383a/src/puppetlabs/puppetdb/scf/migrate.clj#L1891). We could either generate the list using the same date range logic, or regex match the `:table_name`. I would guess that generating the table names with the range logic is likely simpler.

Reply:
Oh wow yeah that's a problem! Where is that date range logic? It looks like in that test we are only inserting a single report so why are there multiple days or report table partitions?

Reply:
it's here https://github.com/puppetlabs/puppetdb/blob/10435954d814f737a94752d264595d4a7e88383a/src/puppetlabs/puppetdb/scf/migrate.clj#L1891